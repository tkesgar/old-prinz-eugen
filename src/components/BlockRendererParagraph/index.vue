<template>
  <p v-html="html" />
</template>

<script>
export default {
  props: {
    data: {
      type: Object,
      required: true
    }
  },
  computed: {
    html () {
      const containerEl = document.createElement('p')
      containerEl.innerHTML = this.data.text

      containerEl.querySelectorAll('a').forEach(anchorEl => {
        const url = encodeURIComponent(anchorEl.href)
        anchorEl.href = `/go?url=${url}`
      })

      return containerEl.innerHTML
    }
  }
}
</script>
