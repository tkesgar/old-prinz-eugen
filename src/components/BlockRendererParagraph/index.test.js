import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import BlockRendererParagraph from './'

describe('BlockRendererParagraph', () => {
  test('should match snapshot', () => {
    const wrapper = mount(BlockRendererParagraph, {
      localVue,
      propsData: {
        data: {
          text: '*Yawns* ... Battleships? Nothing but small fry.'
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render html', () => {
    const wrapper = mount(BlockRendererParagraph, {
      localVue,
      propsData: {
        data: {
          text: '<i>Yawns</i> ... <b>Battleships<b>? Nothing but <u>small fry</u>.'
        }
      }
    })

    expect(wrapper.find('i').exists()).toBe(true)
    expect(wrapper.find('b').exists()).toBe(true)
    expect(wrapper.find('u').exists()).toBe(true)
  })
})
