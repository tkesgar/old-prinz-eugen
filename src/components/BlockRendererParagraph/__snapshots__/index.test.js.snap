// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRendererParagraph should match snapshot 1`] = `
<p>
  *Yawns* ... Battleships? Nothing but small fry.
</p>
`;
